lib/web_ui/lib/src/engine/dom_renderer.dart

```diff
@@ -220,18 +220,28 @@ class DomRenderer {
     _styleElement = html.StyleElement();
     html.document.head.append(_styleElement);
     final html.CssStyleSheet sheet = _styleElement.sheet;
-
+    final bool isWebKit = browserEngine == BrowserEngine.webkit;
+    final bool isFirefox = browserEngine == BrowserEngine.firefox;
     // TODO(butterfly): use more efficient CSS selectors; descendant selectors
     //                  are slow. More info:
     //
     //                  https://csswizardry.com/2011/09/writing-efficient-css-selectors/
 
     // This undoes browser's default layout attributes for paragraphs. We
     // compute paragraph layout ourselves.
-    sheet.insertRule('''
-flt-ruler-host p, flt-scene p {
-  margin: 0;
-}''', sheet.cssRules.length);
+    if (isFirefox) {
+      // For firefox set line-height, otherwise textx at same font-size will
+      // measure differently in ruler.
+      sheet.insertRule(
+          'flt-ruler-host p, flt-scene p '
+          '{ margin: 0; line-height: 100%;}',
```

Comment:
Does it break things if we add this to other browsers?

Reply:
Extra CSS matchers slow down browser style recalculation. We should avoid them whenever possible.

Reply:
We are already adding the same CSS matcher to other browser, just without `line-height`.

Reply:
Ah, you're right. Doesn't matter in this case then.